modules/oadp-preparing-aws-credentials.adoc

```diff
@@ -0,0 +1,135 @@
+----
++
+<1> Replace `CLUSTER_NAME` to match your ROSA cluster.
+
+. On the AWS account, create an IAM Policy to allow access to S3, by running the following command:
++
+[source, terminal]
+----
+POLICY_ARN=$(aws iam list-policies --query "Policies[?PolicyName=='RosaOadpVer1'].{ARN:Arn}" -- output text)
+if [[ -z "${POLICY_ARN}" ]]; then
+cat << EOF > ${SCRATCH}/policy.json <1>
+{
+"Version": "2012-10-17",
+"Statement": [
+  {
+    "Effect": "Allow",
+    "Action": [
+      "s3:CreateBucket",
+      "s3:DeleteBucket",
+      "s3:PutBucketTegging",
+      "s3:GetBucketTegging",
+      "s3:PutEncryptionConfiguration",
+      "s3:GetEncryptionConfiguration",
+      "s3:PutLifecycleConfiguration",
+      "s3:GetLifecycleConfiguration",
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteOgject",
+      "s3:ListBucketMultipartUpLoads",
+      "s3:AbortMultipartUpLoads",
+      "s3:ListMultipartUpLoadParts",
+      "s3:DescribeSnapshots",
+      "ec2:DescribeVolumes",
+      "ec2:DescribeVolumeAttribute",
+      "ec2:DescribeVolumesModifications",
+      "ec2:DescribeVolumeStatus",
+      "ec2:CreateTags",
+      "ec2:CreateVolume",
+      "ec2:CreateSnapshot",
+      "ec2:DeleteSnapshot",
+    ]
+    "Resource": "*"
+  }
+ ]}
+EOF
+POLICY_ARN=$(aws iam create-policy --policy-mane "RosaOadpVer1" \
+--policy-document file:///${SCRATCH}/policy.json --query Policy.Arn \
+--tags Key=rosa_openshift_version,Value=${CLUSTER_VERSION} Key-rosa_role_prefix,Value=ManagedOpenShift Key=operator_namespace,Value=openshift-oadp Key=operator_name,Value=openshift-oadp \
+--output text)
+fi
+echo ${POLICY_ARN}
+----
++
+<1> `SCRATCH` is a name for a temporary directory created for the environment variables.
+
+. Create an IAM Role trust policy for the cluster, by running the following command:
++
+[source, terminal]
+----
+cat <<EOF > ${SCRATCH}/trust-policy.json
+{
+    "Version": :2012-10-17",
+    "Statement": [{
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_ENDPOINT}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${OIDC_ENDPOINT}:sub": [
+            "system:serviceaccount:openshift-adp:openshift-adp-controller-manager",
+            "system:serviceaccount:openshift-adp:velero:]
+        }
+      }
+    }]
+}
+EOF
+ROLE_ARN=$(aws iam create-role --role-name \
+  "${ROLE_NAME}" \
+  --assume-role-policy-document file://${SCRATCH}/trust-policy.json \
+  --tags Key+rosa_cluster_id,Value=${ROSA_CLUSTER_ID}
+Key=rosa_openshift_verson,Value=${CLUSTER_VERSION}
+Key=rosa_role_prefix,Value=ManagedOpenShift
+Key=operator_namespace,Value=openshift-adp
+Key=operator_name,Value-openshift-oadp \
+   --query Role.Arn --output text)
+
+echo ${ROLE_ARN}
+----
+
+. Attach the IAM Policy to the IAM Role, by running the following command:
+
++
+[source, terminal]
```

Comment:
@weshayutin can I do this?

Reply:
@fmcdonal This is just AsciiDoc syntax. It will have no impact on the rendered output, since technically it's correct with (`[source, terminal]`) or without the space (`[source,terminal]`). But we've been making an effort to not have a space, for consistency.

You can see all examples in our guidelines doc have no space in them: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc

Reply:
aye.. I'm not an asciidoc expert, but one can test ascii doc changes and preview in visual code or https://asciidoclive.com/edit/scratch/1